### PR TITLE
Distinct state retry

### PR DESCRIPTION
**Pull request by tiemevanveen**
Retry after https://github.com/omnidan/redux-undo/pull/59

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -315,12 +315,6 @@ export function isHistory (history) {
 }
 // /isHistory
 
-// distinctState helper
-export function distinctState () {
-  return (action, currentState, previousState) => currentState !== previousState
-}
-// /distinctState
-
```
Comment (tiemevanveen) [inline src/index.js]: Sad but true :musical_note:

File: test/index.spec.js [hunk 2]
```diff
@@ -54,45 +54,51 @@ runTestWithConfig(testConfigThree, initialStateThree, 'Erroneous configuration')
...
 
       mockInitialState = mockUndoableReducer(undefined, {})
       incrementedState = mockUndoableReducer(mockInitialState, { type: 'INCREMENT' })
-      doubleIncrementedState = mockUndoableReducer(incrementedState, { type: 'INCREMENT' })
       console.info('  Beginning Test! Good luck!')
       console.info('    initialStoreState:     ', initialStoreState)
       console.info('    store.getState():      ', store.getState())
       console.info('    mockInitialState:      ', mockInitialState)
       console.info('    incrementedState:      ', incrementedState)
-      console.info('    doubleIncrementedState:', doubleIncrementedState)
```
Comment (tiemevanveen) [inline test/index.spec.js]: Moved to test

File: test/index.spec.js [hunk 3]
```diff
@@ -116,7 +122,35 @@ function runTestWithConfig (testConfig, initialStoreState, label) {
...
+        // swap back for other tests
+        store.replaceReducer(mockUndoableReducer)
+
+        // Increment and check result again
+        expectedResult = countReducer(store.getState().present, {type: 'INCREMENT'})
+        store.dispatch({type: 'INCREMENT'})
+        expect(store.getState().present).to.equal(expectedResult)
+      })
+    })
+
```
Comment (tiemevanveen) [inline test/index.spec.js]: I think this is the behavior that we're looking for right?

File: test/index.spec.js [hunk 4]
```diff
@@ -154,8 +193,9 @@ function runTestWithConfig (testConfig, initialStoreState, label) {
       })
 
       it('should increment when action is dispatched to store', () => {
+        let expectedResult = store.getState().present + 1
         store.dispatch({type: 'INCREMENT'})
-        expect(store.getState()).to.deep.equal(incrementedState)
+        expect(store.getState().present).to.equal(expectedResult)
```
Comment (tiemevanveen) [inline test/index.spec.js]: We can't rely on `incrementedState` here, We're going through the store and history might have changed by other tests. This is kindof a bonus test so it's fine here

File: test/index.spec.js [hunk 5]
```diff
@@ -327,6 +367,7 @@ function runTestWithConfig (testConfig, initialStoreState, label) {
       let doubleUndoState
       let doubleRedoState
       before('perform a jump action', () => {
+        let doubleIncrementedState = mockUndoableReducer(incrementedState, { type: 'INCREMENT' })
```
Comment (tiemevanveen) [inline test/index.spec.js]: Moved

File: test/index.spec.js (hunk 3 above)
Comment (pl12133) [inline test/index.spec.js]: That looks correct to me.

File: test/index.spec.js (hunk 4 above)
Comment (pl12133) [inline test/index.spec.js]: If this becomes an issue in the long term we can re-initialize the store in-between tests with:
```js
  before('reinitialize store', () => {
    store = Redux.createStore(...)
  })
```

File: src/index.js (hunk 1 above)
Comment (pl12133) [inline src/index.js]: :+1:

File: test/index.spec.js (hunk 4 above)
Comment (tiemevanveen) [inline test/index.spec.js]: Yeah.. I considered to do that but found this too small / bonus that I didn't do it.

**Pull request merged**